python/understack-workflows/understack_workflows/main/synchronize_obm_creds.py

```diff
@@ -0,0 +1,105 @@
+import json
+import sys
+
+import ironicclient.common.apiclient.exceptions
+
+from understack_workflows.helpers import setup_logger
+from understack_workflows.ironic.client import IronicClient
+from understack_workflows.ironic.secrets import read_secret
+from understack_workflows.node_configuration import IronicNodeConfiguration
+
+logger = setup_logger(__name__)
+
+
+def event_to_node_configuration(event: dict) -> IronicNodeConfiguration:
+    node_config = IronicNodeConfiguration()
+    node_config.conductor_group = None
+    node_config.driver = "redfish"
+
+    node_config.chassis_uuid = None
+    node_config.uuid = event["device"]["id"]
+    node_config.name = event["device"]["name"]
+
+    return node_config
+
+
+def credential_secrets():
+    """Reads Kubernetes Secret files with username/password credentials."""
+    username = None
+    password = None
+    with open("/etc/obm/username") as f:
+        # strip leading and trailing whitespace
+        username = f.read().strip()
+
+    with open("/etc/obm/password") as f:
+        # strip leading and trailing whitespace
+        password = f.read().strip()
+
+    return [username, password]
+
+
+def replace_or_add_field(path, current_val, expected_val):
+    if current_val == expected_val:
+        return None
+    if current_val is None:
+        return {"op": "add", "path": path, "value": expected_val}
+    else:
+        return {"op": "replace", "path": path, "value": expected_val}
+
+
+def main():
+    if len(sys.argv) < 1:
+        raise ValueError(
+            "Please provide node configuration in JSON format as first argument."
+        )
+
+    logger.info("Pushing device new node to Ironic.")
+    client = IronicClient(
+        svc_url=read_secret("IRONIC_SVC_URL"),
+        username=read_secret("IRONIC_USERNAME"),
+        password=read_secret("IRONIC_PASSWORD"),
+        auth_url=read_secret("IRONIC_AUTH_URL"),
+        tenant_name=read_secret("IRONIC_TENANT"),
+    )
```

Comment:
this code is duplicated in at least one another workflow. Leaving note here so that we remember to extract it later in separate PR. Feel free to mark this as resolved for now.

Reply:
Oh yes. There's quite a few spots that need to be de-duped.